exams/pearson_test.py

```diff
@@ -0,0 +1,214 @@
+            'profile__address2': 'Room B345',
+            'profile__city': 'Boston',
+            'profile__state_or_territory': 'Massachusetts',
+            'profile__country': 'US',
+            'profile__postal_code': '02115',
+            'profile__phone_number': '999-999-9999',
+            'profile__phone_country_code': '1',
+        }
+
+        with mute_signals(post_save):
+            exam_profiles = [ExamProfileFactory.create(**kwargs)]
+
+        exam_profiles[0].profile.updated_on = datetime(2016, 5, 15, 15, 2, 55)
+
+        write_cdd_file(file, exam_profiles)
+
+        lines = file.getvalue().splitlines()
+
+        row = lines[1]
+
+        assert row == (
+            "14879\tJane\tSmith\tjane@example.com\t"
+            "1 Main St\tRoom B345\t\t"  # double tab is for blank address3
+            "Boston\tMassachusetts\t02115\tUSA\t"
+            "999-999-9999\t1\t2016/05/15 15:02:55"
+        )
+
+    def test_write_cdd_file_skips_invalid_state(self):  # pylint: disable=no-self-use
+        """
+        Tests write_cdd_file against a profiel with invalid state
+        """
+        file = io.StringIO()
+
+        with mute_signals(post_save):
+            profiles = [ExamProfileFactory.create(profile=ProfileFactory.create(country='XXXX'))]
+
+        write_cdd_file(file, profiles)
+
+        lines = file.getvalue().splitlines()
+
+        _, rows = lines[0], lines[1:]
+
+        assert len(rows) == 0
+
+    def test_upload_tsv(self):  # pylint: disable=no-self-use
+        """
+        Tests that upload uses the correct settings values
+        """
+        EXAMS_SFTP_HOST = 'l0calh0st'
+        EXAMS_SFTP_PORT = '345'
+        EXAMS_SFTP_USERNAME = 'username'
+        EXAMS_SFTP_PASSWORD = 'password'
+        EXAMS_SFTP_UPLOAD_DIR = 'tmp'
+        FILENAME = 'FILENAME'
+
+        with self.settings(
+            EXAMS_SFTP_HOST=EXAMS_SFTP_HOST,
+            EXAMS_SFTP_PORT=EXAMS_SFTP_PORT,
+            EXAMS_SFTP_USERNAME=EXAMS_SFTP_USERNAME,
+            EXAMS_SFTP_PASSWORD=EXAMS_SFTP_PASSWORD,
+            EXAMS_SFTP_UPLOAD_DIR=EXAMS_SFTP_UPLOAD_DIR
+        ), patch('pysftp.Connection') as connection_mock:
+            from exams.pearson import upload_tsv
+
+            upload_tsv(FILENAME)
+            connection_mock.assert_called_once_with(
+                host=EXAMS_SFTP_HOST,
+                port=int(EXAMS_SFTP_PORT),
+                username=EXAMS_SFTP_USERNAME,
+                password=EXAMS_SFTP_PASSWORD,
+                cnopts=mock.ANY,
+            )
+
+            ftp_mock = connection_mock.return_value.__enter__.return_value
+            ftp_mock.cd.assert_called_once_with(EXAMS_SFTP_UPLOAD_DIR)
+            ftp_mock.put.assert_called_once_with(FILENAME)
+
+    def test_upload_tsv_raise_improperly_configured(self):  # pylint: disable=no-self-use
+        """
+        Tests that upload raises ImproperlyConfigured if setting is missing
+        """
+
+        with self.settings(
```

Comment:
Can you break these into separate tests, to make it easier to identify if a particular case failed?